src/mono/mono/mini/interp/transform.c

```diff
@@ -8334,7 +8334,7 @@ interp_cprop (TransformData *td)
 				local_defs [dreg].def_index = ins_index;
 			}
 
-			if (opcode == MINT_MOV_4 || opcode == MINT_MOV_8 || opcode == MINT_MOV_VT) {
+			if (opcode == MINT_MOV_4 || opcode == MINT_MOV_8 || (opcode == MINT_MOV_VT && td->locals [sregs [0]].size == td->locals [dreg].size)) {
 				int sreg = sregs [0];
```

Comment:
Shouldn't this case use a separate opcode instead ?

Reply:
changed the code so we don't emit partial vt moves with MOV_VT. We actually a separate dummy opcode for situations like this